digest.go

```diff
@@ -0,0 +1,272 @@
+package resty
+
+import (
+	"crypto/md5"
+	"crypto/rand"
+	"crypto/sha256"
+	"crypto/sha512"
+	"errors"
+	"fmt"
+	"hash"
+	"io"
+	"net/http"
+	"strings"
+)
+
+var (
+	ErrBadChallenge    = errors.New("challenge is bad")
+	ErrCharset         = errors.New("unsupported charset")
+	ErrAlgNotSupported = errors.New("algorithm is not supported")
+	ErrQopNotSupported = errors.New("no supported qop in list")
+	ErrNoQop           = errors.New("qop must be specified")
+)
```

Comment:
@segevda Can you please update Error names to following?
```go
var (
	ErrDigestBadChallenge    = errors.New("digest: challenge is bad")
	ErrDigestCharset         = errors.New("digest: unsupported charset")
	ErrDigestAlgNotSupported = errors.New("digest: algorithm is not supported")
	ErrDigestQopNotSupported = errors.New("digest: no supported qop in list")
	ErrDigestNoQop           = errors.New("digest: qop must be specified")
)
```

Reply:
Done